### PR TITLE
feat: user-provided sysroot

### PR DESCRIPTION
**Pull request by f0rmiga**
This adds the possibility of users providing a sysroot different than the one shipped with the bootlin sysroots.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -16,30 +16,72 @@ load("@aspect_bazel_lib//lib:repositories.bzl", "aspect_bazel_lib_dependencies")
...
+)
+""",
+    sha256 = "84656a6df544ecef62169cfe3ab6e41bb4346a62d3ba2a045dc5a0a2ecea94a3",
+    urls = ["https://commondatastorage.googleapis.com/chrome-linux-sysroot/toolchain/2202c161310ffde63729f29d27fe7bb24a0bc540/debian_stretch_amd64_sysroot.tar.xz"],
+)
+
 load("//toolchain:defs.bzl", "gcc_register_toolchain")
 
-flags = [
+common_flags = [
```
Comment (alexeagle) [inline WORKSPACE:36]: would we want users to copy-paste this code and have a WORKSPACE file like this? I imagine we'd want them to break out some starlark library files, so then we should do that too for illustration

File: WORKSPACE [hunk 2]
```diff
@@ -16,30 +16,72 @@ load("@aspect_bazel_lib//lib:repositories.bzl", "aspect_bazel_lib_dependencies")
...
     "-Wall",
     "-Wextra",
     "-fdiagnostics-color=always",
 ]
 
+ldflags = common_flags + [
+    "-B{sysroot}/usr/lib/x86_64-linux-gnu",
```
Comment (alexeagle) [inline WORKSPACE:43]: nit: who interpolates `{sysroot}` and are other variables available as well? a bit of docs would be nice

File: WORKSPACE [hunk 3]
```diff
@@ -16,30 +16,72 @@ load("@aspect_bazel_lib//lib:repositories.bzl", "aspect_bazel_lib_dependencies")
...
 load("//toolchain:defs.bzl", "gcc_register_toolchain")
 
-flags = [
+common_flags = [
     "-Wall",
     "-Wextra",
     "-fdiagnostics-color=always",
 ]
 
+ldflags = common_flags + [
```
Comment (alexeagle) [inline WORKSPACE:42]: nit: I think it's easier to reason about if you add `common_flags` to all or none of these other flag groups. currently this is the only one that doesn't get the `common_flags` added to it later when this symbol is referenced

File: examples/hello_world_c/BUILD.bazel [hunk 4]
```diff
@@ -1,4 +1,33 @@
+load("@aspect_bazel_lib//lib:transitions.bzl", "platform_transition_filegroup")
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
+
 cc_binary(
     name = "hello_world_c",
     srcs = ["main.c"],
 )
+
+platform_transition_filegroup(
```
Comment (alexeagle) [inline examples/hello_world_c/BUILD.bazel:9]: nit: the examples are more useful if they have some comments

**Review by alexeagle — APPROVED** [review]

File: examples/hello_world_c/BUILD.bazel (hunk 4 above)
Comment (f0rmiga) [reply examples/hello_world_c/BUILD.bazel:9]: I'm reworking the examples as tests in another branch, so this is only a temporary assertion.

File: WORKSPACE (hunk 1 above)
Comment (f0rmiga) [reply WORKSPACE:36]: I can clean this all up in the follow-up PR where I'm reworking the examples.

File: WORKSPACE (hunk 2 above)
Comment (f0rmiga) [reply WORKSPACE:43]: The doc already exists in the attribute.

File: WORKSPACE (hunk 3 above)
Comment (f0rmiga) [reply WORKSPACE:42]: Same re cleaning up in the examples re-work I'm doing.

**Pull request merged**